softlayer/resource_softlayer_lb_local_service.go

```diff
@@ -208,18 +228,37 @@ func resourceSoftLayerLbLocalServiceUpdate(d *schema.ResourceData, meta interfac
 
 	log.Println("[INFO] Updating load balancer service")
 
-	success, err := services.GetNetworkApplicationDeliveryControllerLoadBalancerVirtualIpAddressService(sess).
-		Id(vipID).
-		EditObject(&vip)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"pending"},
+		Target:  []string{"complete"},
+		Refresh: func() (interface{}, string, error) {
+			_, err := services.GetNetworkApplicationDeliveryControllerLoadBalancerVirtualIpAddressService(sess).
+				Id(vipID).
+				EditObject(&vip)
+
+			if apiErr, ok := err.(sl.Error); ok {
+				// 500 could mean that the LB is busy with another transaction. Retry
+				if apiErr.StatusCode == 500 {
+					return false, "pending", nil
+				}
+
+				// Any other error is unexpected. Abort
+				return false, "", err
+			}
+
+			return true, "complete", nil
+		},
+		Timeout:    10 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
```

Comment:
seems like you could put this in a function so that it can be reused from create and update.